### PR TITLE
refactor: DeploymentName to DeploymentKey

### PR DESCRIPTION
**Pull request by matt2e**
Separate PR on top of other PR, to separate them out for review

This refactors everything to refer to a deployment's key instead of its name, including:
- db
- events
- console

### CODE REVIEW

**Comment by alecthomas** [comment]
This is a big change, which is expected, but can you run the integration tests locally please? `integration-tests`

**Review by alecthomas — APPROVED** [review]
LGTM just one comment re. running integration tests.

File: backend/controller/controller.go [hunk 1]
```diff
@@ -285,14 +285,14 @@ func (s *Service) Status(ctx context.Context, req *connect.Request[ftlv1.StatusR
 	deployments, err := slices.MapErr(status.Deployments, func(d dal.Deployment) (*ftlv1.StatusResponse_Deployment, error) {
 		labels, err := structpb.NewStruct(d.Labels)
 		if err != nil {
-			return nil, fmt.Errorf("could not marshal attributes for deployment %s: %w", d.Name, err)
+			return nil, fmt.Errorf("could not marshal attributes for deployment %s: %w", d.Key.String(), err)
```
Comment (alecthomas) [inline backend/controller/controller.go:288]: FYI you don't need to use `.String()` with `%s` - the formatter will do this automatically.

Comment (matt2e) [reply backend/controller/controller.go:288]: Oh neat

**Comment by matt2e** [comment]
> This is a big change, which is expected, but can you run the integration tests locally please? `integration-tests`

All tests passing 👍

**Pull request merged**